libraries/botbuilder-lg/src/activityFactory.ts

```diff
@@ -0,0 +1,361 @@
+    }
+
+    private static buildActivityFromLGStructuredResult(lgJObj: any): Activity {
+        let activity: any = {};
+
+        const type: string = this.getStructureType(lgJObj);
+        if (ActivityFactory.GenericCardTypeMapping.has(type)) {
+            const attachment: Attachment = this.getAttachment(lgJObj);
+            if (attachment !== undefined) {
+                activity = MessageFactory.attachment(attachment);
+            }
+        } else if (type === 'activity') {
+            activity = this.buildActivityFromObject(lgJObj);
+        } else {
+            throw new Error(`type ${ type } is not support currently.`);
+        }
+
+        return activity;
+    }
+
+    private static buildActivityFromObject(lgJObj: any): Activity {
+        let activity: any = {};
+
+        if ('type' in lgJObj && lgJObj.type === 'event') {
+            activity = this.buildEventActivity(lgJObj);
+        } else {
+            activity = this.buildMessageActivity(lgJObj);
+        }
+
+        return activity;
+    }
+
+    private static buildEventActivity(lgJObj: any): Activity {
+        let activity: any = { type: ActivityTypes.Event };
+        for (const item of Object.keys(lgJObj)) {
+            const property: string = item.trim();
+            const value: any = lgJObj[item];
+            switch (property.toLowerCase()) {
+                case '$type':
+                    break;
+                case 'name':
+                    activity.name = value.toString();
+                    break;
+                case 'value':
+                    activity.value = value.toString();
+                    break;
+                default:
+                    console.log(`Skipping unknown activity property ${ property }`);
+                    break;
+            }
+        }
+
+        return activity;
+    }
+
+    private static buildMessageActivity(lgJObj: any): Activity {
+        let activity: any = { type: ActivityTypes.Message };
+        for (const key of Object.keys(lgJObj)) {
+            const property: string = key.trim();
+            const value: any = lgJObj[key];
+
+            switch (property.toLowerCase()) {
+                case '$type':
+                    break;
+                case 'text':
+                    activity.text = value.toString();
+                    break;
+                case 'speak':
+                    activity.speak = value.toString();
+                    break;
+                case 'inputhint':
+                    activity.inputHint = value.toString();
+                    break;
+                case 'attachments':
+                    activity.attachments = this.getAttachments(value);
+                    break;
+                case 'suggestedactions':
+                    activity.suggestedActions = this.getSuggestions(value);
+                    break;
+                case 'attachmentlayout':
+                    activity.AttachmentLayout = value.toString();
```

Comment:
>AttachmentLayout [](start = 29, length = 16)

Case is wrong... That's another reason you want activity to be strongly typed and not an any